invenio_app_ils/items/api.py

```diff
@@ -60,16 +58,19 @@ def ensure_item_can_be_updated(self, record):
         """Raises an exception if the item's status cannot be updated."""
         latest_version = record.revisions[-1]
         if latest_version:
-            status = latest_version.get("status", None)
-        else:
-            status = None
-        pid = record["pid"]
-        if status == "CAN_CIRCULATE":
+            latest_version_document_pid = latest_version.get("document_pid", None)
+        document_changed = latest_version_document_pid != record.get("document_pid", None)
+        if document_changed:
+            pid = record["pid"]
             item_pid = dict(value=pid, type=ITEM_PID_TYPE)
-            active_loan = get_active_loan_by_item_pid(item_pid).execute().hits
-            total = active_loan.total if lt_es7 else active_loan.total.value
-            if total > 0:
-                raise ItemHasActiveLoanError(active_loan[0]["pid"])
+            if search_by_pid(
+                item_pid=item_pid,
+                filter_states=current_app.config["CIRCULATION_STATES_LOAN_ACTIVE"]
```

Comment:
Note for others: as of now this is equivalent to searching for non pending loans.